api/src/server/index.js

```diff
@@ -59,29 +59,27 @@ const createServer = ({
 			 */
 			db: database,
 
-			/**
-			 * Returns the id of the currently logged in user, or `null` otherwise.
-			 * Beware: this doesn’t guarantee you that the user still exists in the DB!
-			 */
-			getUserId: () => req.session.principal ? req.session.principal.id : null,
-
 			/**
 			 * Returns the id of the currently logged in user, or throws an error otherwise.
 			 * Beware: this doesn’t guarantee you that the user still exists in the DB!
 			 */
-			getUserIdOrThrow: () => {
-				if (req.session.principal && req.session.principal.id)
-					return req.session.principal.id
-				throw new AuthenticationError('No user is logged in.')
+			getCurrentUserId: () => {
+				if (!req.session.principal || !req.session.principal.id)
```

Comment:
I flipped this check to make it inline with the new one below.

Reply:
Looks good. Seems a bit like double code - it'd be cleaner to call the function getCurrentUserId inside getCurrentUser - but it's a tiny thing so I don't really care. All fine. I'll merge it in.

Reply:
That’s a good point and I would have also preferred that. We’d need to pull it out then, unfortunately, because we can’t access `getCurrentUserId` while we are still in the scope of the object literal.

```js
// Create a new object via an object literal
const foo = {
  bar: () => console.log(123),
  asdf: this.bar(), // There `this` in object literals, so `this` refers to the global object here
  qwer: bar(), // Syntax error, `bar` is undefined
}
```

I thought pulling out a separate helper method makes the code less cohesive, so I decided to go with duplication here. It’s not ideal either way, though…

Reply:
Yeah, it's not a big thing. So let's leave it. :-)